components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowManagementService.java

```diff
@@ -378,4 +378,14 @@ WorkflowRequest[] getRequestsFromFilter(String user, String beginDate, String en
 
     List<String> listEntityNames(String wfOperationType, String wfStatus, String entityType, int tenantID, String
             idFilter) throws WorkflowException;
+            
+    /**
+     * Retrieve workflow request from request ID
+     *
+     * @param requestId request id.
+     * @return
+     * @throws WorkflowException
+     */
+    WorkflowRequest getWorkflowRequest(String requestId) throws WorkflowException;
```

Comment:
Let's use the default implementation approach.

default WorkflowRequest getWorkflowRequest(String requestId) throws WorkflowException {

        throw new NotImplementedException("getRoles method is not implemented");
    }